external/fv3fit/fv3fit/emulation/models/microphysics.py

```diff
@@ -155,3 +155,143 @@ def build(self, data: Mapping[str, tf.Tensor],) -> tf.keras.Model:
                 **self._get_residual_outputs(inputs, data, hidden),
             },
         )
+
+
+@dataclasses.dataclass(frozen=True)
+class Field:
```

Comment:
This way of configuring avoids the coupling between `Microphysics.{input_variables,residual_out_variables, etc}`. The keys, values of those configurations have various constraints that make it easy to make an invalid config.

We could probably modify `MicrophysicsConfig` to use `List[Field]` instead of those configurations.

Reply:
Yeah, this is a nice field definition that I think would work well for simplifying `MicrophysicsConfig`.